IBM/ibm-common-service-operator
# Add global hugepage support for IM and EDB

issue: https://github.ibm.com/IBMPrivateCloud/roadmap/issues/62225

### Context
To provide a global hugepage setting configuration via CommonService CR interface for IM and EDB.

### Interface
Create or update arbitrary CommonService CR
- configure `.spec.hugepages.enable: true`, to enable hugepages setting for CS
- configure one or more `.spec.hugepages.huagepages-<size>: <allocation>`, to specify the hugepages size and allocation
   - both `<size>` and `<allocation>` should be in format of Quantity. e.g., `1Mi`, `1Gi` and so on.
   - If `<allocation>` is not specified, then default `100Mi` is specified by CS

```
apiVersion: operator.ibm.com/v3
kind: CommonService
metadata:
  name: huge-pages-common-service
spec:
  hugepages:
    enable: true
    hugepages-1Gi: 500Mi
    hugepages-2Gi: 2Mi
    hugepages-2Mi: ''
```

How to test
- Patch CS operator CSV with image `docker-na-public.artifactory.swg-devops.com/hyc-cloud-private-scratch-docker-local/ibmcom/common-service-operator-amd64:dev` and change ImagePullPolicy to `Always`
- Update CS CR with following configurations
    ```yaml
    apiVersion: operator.ibm.com/v3
    kind: CommonService
    metadata:
      name: huge-pages-common-service
    spec:
      hugepages:
        enable: true
        hugepages-1Gi: 500Mi
        hugepages-2Gi: 2Mi
        hugepages-2Mi: ''
    ```
- Check if OperandConfig is updated with the same configuration for IM and Common Service PostgreSQL

    ```yaml
        - name: ibm-im-operator
          spec:
            authentication:
              authService:
                resources:
                  limits:
                    cpu: 1000m
                    hugepages-1Gi: 500Mi
                    hugepages-2Gi: 2Mi
                    hugepages-2Mi: 100Mi
                    memory: 1090Mi
                  requests:
                    cpu: 600m
                    ephemeral-storage: 256Mi
                    memory: 650Mi
    ```
